packages/wonder-blocks-tooltip/components/tooltip.js

```diff
@@ -80,25 +81,36 @@ export default class Tooltip extends React.Component<Props, State> {
         }
     }
 
+    _renderPopper(active: boolean) {
+        if (!active) {
+            return null;
+        }
+
+        const {placement} = this.props;
+        return (
+            <TooltipPopper
+                anchorElement={this.state.anchorElement}
+                placement={placement || Tooltip.defaultProps.placement}
+            >
+                {(props) => (
+                    <TooltipBubble popperProps={props}>
+                        {this._renderBubbleContent()}
+                    </TooltipBubble>
+                )}
+            </TooltipPopper>
+        );
+    }
+
     render() {
-        const {forceAnchorFocusivity, placement} = this.props;
+        const {forceAnchorFocusivity} = this.props;
         return (
             <TooltipAnchor
                 forceAnchorFocusivity={forceAnchorFocusivity}
                 anchorRef={(r) => this._updateAnchorElement(r)}
             >
                 {(active) => (
                     <TooltipPortalMounter anchor={this._renderAnchorElement()}>
-                        {active ? (
-                            <TooltipBubble
-                                placement={
-                                    placement || Tooltip.defaultProps.placement
-                                }
-                                anchorElement={this.state.anchorElement}
-                            >
-                                {this._renderBubbleContent()}
-                            </TooltipBubble>
-                        ) : null}
+                        {this._renderPopper(active)}
```

Comment:
It would be good to document the full hierarchy somewhere, maybe at the top of this render function.  Is it?
```
- TooltipAnchor
-- TooltipPortalMounter
--------------------- portal divide
- TooltipPopper
-- TooltipArrow
-- TooltipBubble
--- TooltipContent
```

Reply:
Yes, I agree. Hadn't done that yet since we weren't quite finished, but that's a great thing to include now.

It's:

```
- TooltipAnchor (Behavioral)
-- TooltipPortalMounter (Behavioral)
--------------------------- portal divide
- TooltipPopper (Behavioral)
-- TooltipBubble (Render)
--- TooltipContent (Render)
--- TooltipArrow (Render)
```